Improvements for Default color button in filament settings

yw4z opened the pull request:
tested on windows 11

### CHANGES
• Matches width with other parameters on all platforms
• Adds a function for resetting value system default (no color / transparent) with right click
• Matches UI style of button on Windows
---- • Matched style with buttons and used "Pick ..." as label. used 3 dots because its opens a dialog like on other buttons
---- • Added support for focus border and hover effects

### COMPARISON (WINDOWS ONLY)
BEFORE-AFTER - No color chosen
![Screenshot-20250508210022](https://github.com/user-attachments/assets/d2e4c1c8-d179-41ae-89e4-8dbd0c44dbfb)
![Screenshot-20250508205957](https://github.com/user-attachments/assets/f9e001fe-92f2-4180-9a61-f6eaa93cca52)

BEFORE-AFTER - Color chosen
![Screenshot-20250508210033](https://github.com/user-attachments/assets/937601ff-de39-4d04-8bef-0ff287e6283a)
![Screenshot-20250508210048](https://github.com/user-attachments/assets/2c1d621d-d889-4d98-ace5-7c5c234c3de9)

Noisyfox commented:
![image](https://github.com/user-attachments/assets/9d5ffbd4-a154-4b90-9ff7-eafb2ae72f45)

Unfortunately this does not work on macOS

yw4z commented:
can you click on it and is that changes its color after picking a color? maybe switching to wxPaintDc will fix that

Noisyfox commented:
![image](https://github.com/user-attachments/assets/136a766e-636d-40d5-aa72-d41a10afba3d)
![image](https://github.com/user-attachments/assets/d99561b2-096b-404e-beb3-3b771a41d129)
Works perfectly on Ubuntu.

Noisyfox commented:
> can you click on it and is that changes its color after picking a color?

No it didn't. I've already picked a color in the screenshot .

yw4z commented:
wow didnt know that ubuntu color picker look like that. i assume this requires a custom class so i'm closing it for now if you dont have any recommendation

yw4z commented:
alternatively i can limit bitmap button usage (this PR) with windows since its ugliest one

Noisyfox commented:
> alternatively i can limit bitmap button usage (this PR) with windows since its ugliest one

I think it's already looking good enough on macOS and Linux, so I think this is a good idea.

yw4z commented:
updated code only work for windows
but added 2 things for all platforms
• matched width of button with other parameter boxes
• added a function to reset value to default with right click. this was not possible on windows before. not sure about other platforms. also added an information to tooltip for this behavior

Noisyfox reviewed (approved):
LGTM, thanks
